src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopies = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        stateCopies.push({ ...stateCopy });
```

Comment:
You don't need to repeat this line, just write it below once 🙂

Reply:
Agreed, fixed it accordingly, thanks for the review 😃